### ISSUE TITLE
Worldmap: Format/Add/Test Curated Target Layers

### ISSUE THREAD

**Issue by raprasad** [IQSS/dataverse#3477]
### Steps for each layer:
 
1.  Add to WorldMap
1.  Specify as JoinTarget
1.  Test via GeoConnect interface
  1.  correct column format, incorrect column format
    1.  int -> char; char -> int; double -> int; int -> double
    1.  update code as needed

## Layer list

Full checklist at: [google doc](https://docs.google.com/a/harvard.edu/spreadsheets/d/1qumAPAvbVwvXFUxngb9-_FFkIglGwMJm881pE8n6uV4/edit?usp=sharing)

For administrative layers:
  - Data: [Geographical Infrastructure v. 2015 Dataverse](https://dataverse.harvard.edu/dataverse/geographical_infrastructure_2015)
  - Docs: [Documentation for the 2015 Geographical Infrastructure](https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/C5IULB)

### Boston for now, but we want Massachusetts (or National) data

- [x] Census Tracts 2010, Boston 
  - [x] test depends on #3485
- [x] Census Block Groups 2010, Boston
- [x] Census Blocks 2010, Boston
- [x] US Zip Codes 2014, Boston

* Note: ~~__The following admin. layers assume the fix of #3484__~~ done 11/22

### Boston specific data

- [x] BRA Neighborhood Statistical Areas 2015, Boston
- [x] BRA Planning Districts 2015, Boston
- [x] ~~City Council Districts 2014, Boston~~
- [x] Election Precincts 2015, Boston  
- [x] Election Wards 2015, Boston
- [x] Fire Districts 2015, Boston
    - [x] Layer loaded as target
    - [x] *CHECK*: Number are discontinuous: 1, 3, 4, 6, 7, 8, 9, 11, 12.  Correct?
       - ok, same as dataverse archived file
- [x] ISD Neighborhoods 2015, Boston
- [x] Police Districts 2015, Boston
- [x] PWD Districts 2015, Boston
- [x] Addresses 2015, Boston
- [x] Parcels 2015, Boston **moved to #3493**
    - 11/22 - keeps hanging on upload
- [x] Roads 2015, Boston
- [x] Intersections 2015, Boston
  - [x] City intersections
  - [x] Census intersections

**Comment by raprasad** [comment]
Election Precincts note:
 - In original format, precinct id is a "double" in WorldMap, not an "int"--although looks like an int with .00000 added
- Currently double and int not joining.
  - Options:
    - [ ] Try reformat of orig. file?  (Yes, easy 1st option)
    - [ ] Detect WorldMap "double" and do pandas convert (If same issue in other files)

**Comment by raprasad** [comment]
bug:
- [ ] When manually changing tabular file via admin, clear JSON column names attribuate

**Comment by raprasad** [comment]
For fire districts (and possibly BPA planning districts + election wards) there is truncated data. e.g. when downloading the .xls:

check with Ben re: ```DATA TRUNCATED MULTIPOLYGON```
  - May be just message for Excel download.  All data seems to be in postgis

```
DATA TRUNCATED MULTIPOLYGON (((746870.6151123643 2935873.384744346, 746973.7536856085 2935786.328816101, 747212.6042577773 2935584.6510698497, 747286.2008953542 29355
(etc, etc. for many, many lines)
```

**Comment by raprasad** [comment]
Note: updating 3 layers to use version in Dataverse 2015 BARI Regional Geographies: https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/PJ3JK2

**Comment by raprasad** [comment]
census layout (e.g. GEOID10): https://www.census.gov/geo/maps-data/data/tract_rel_layout.html

**Comment by raprasad** [comment]
block groups (bg_id_10 in BARI file) are 12-digits FIPS codes that may have leading zeros:

![fipscode_part4](https://cloud.githubusercontent.com/assets/1036060/20680765/6eac1906-b56e-11e6-89bb-adea0391b7bd.png)

(image source: https://www.policymap.com/blog/2012/08/tips-on-fips-a-quick-guide-to-geographic-place-codes-part-iii/)

**Comment by raprasad** [comment]
Boston zip codes in.  Correct mapping--including error--typo in error message covered in another ticket)

![geoconnect__map_data](https://cloud.githubusercontent.com/assets/1036060/20686613/6dbfaea4-b587-11e6-80d8-96d3594e6fa6.png)

**Comment by raprasad** [comment]
- No actual code to review.  Discussed with @kcondon and moving to QA

- *NOTE*: QA Dependent on configuration/installation of Dev Server.  Target Date 2/9

**Comment by kcondon** [comment]
OK tested various available layers and they worked. Closing.
